pkg/api/v20160330/validate.go

```diff
@@ -71,6 +71,15 @@ func (l *LinuxProfile) Validate() error {
 
 // Validate implements APIObject
 func (a *Properties) Validate() error {
+	if a.OrchestratorProfile == nil {
+		return fmt.Errorf("missing OrchestratorProfile")
+	}
```

Comment:
probably not inline... we have a defaults.go that does stuff like that,but we could skip the validation, though I'm pretty sure other parts of the validation require that we know what orchestrator we are dealing with so we might need to do it inline.